File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let visited=new Array(n).fill(false);
    
    function dfs(k){
        visited[k]=true;
        for(let j=0;j<n;j++){
            if(!visited[j] && computers[k][j]===1){
                dfs(j);
            }
                
        }
    }
    
    for(let i=0;i<n;i++){
        if(!visited[i]){
            answer++;
            dfs(i);
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import defaultdict

def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
                
    def dfs(num):
        nonlocal visited
        for i,c in enumerate(computers[num]):
            if c==1 and visited[i]==False:
                visited[i]=True
                dfs(i)
        
    for i in range(n):
        if(visited[i]==False):
            visited[i]=True
            dfs(i)
            answer+=1
    
    return answer